dcos/mesos.py

```diff
@@ -754,7 +766,15 @@ def dict(self):
         :rtype: dict
         """
 
-        return self._task
+        try:
+            fd = self.fault_domain()
+        except Exception:
+            return self._task
```

Comment:
What exception could be returned here, is it a `KeyError`? I'd rather catch the exact exception inside `fault_domain()` and return `None` in such cases.

Reply:
We shouldn't put try/except blocks around simple KeyError exceptions anyway. We should explicitly check for the existence of the key and return e.g. `None` if appropriate.